extract.c

```diff
@@ -0,0 +1,127 @@
+#include "squashfuse.h"
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include "squashfs_fs.h"
+
+#define PROGNAME "squashfuse_extract"
+
+#define ERR_MISC	(-1)
+#define ERR_USAGE	(-2)
+#define ERR_OPEN	(-3)
+
+static void usage() {
+    fprintf(stderr, "Usage: %s ARCHIVE PATH_TO_EXTRACT\n", PROGNAME);
+    fprintf(stderr, "       %s ARCHIVE -a\n", PROGNAME);
+    exit(ERR_USAGE);
+}
+
+static void die(const char *msg) {
+    fprintf(stderr, "%s\n", msg);
+    exit(ERR_MISC);
+}
+
+
+bool startsWith(const char *pre, const char *str)
+{
+    size_t lenpre = strlen(pre),
+    lenstr = strlen(str);
+    return lenstr < lenpre ? false : strncmp(pre, str, lenpre) == 0;
+}
+
+int main(int argc, char *argv[]) {
+    sqfs_err err = SQFS_OK;
+    sqfs_traverse trv;
+    sqfs fs;
+    char *image;
+    char *path_to_extract;
+    char *prefix;
+    char prefixed_path_to_extract[1024];
+    
+    prefix = "squashfs-root/";
+    
+    if(access(prefix, F_OK ) == -1 ) {
```

Comment:
style nit: space before the open paren of if statements; similar issue in other places